controllers/events.js

```diff
@@ -0,0 +1,171 @@
+const { EventAPIService } = require("../services/EventAPIService");
+const { EventTokenService } = require("../services/EventTokenService");
+const eventQuery = require("../models/events");
+const logger = require("../utils/logger");
+
+const tokenService = new EventTokenService();
+const apiService = new EventAPIService(tokenService);
+
+/**
+ * Creates a new room document in the Firestore database with the data provided in the HTTP request body.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The saved room data in JSON format.
+ * @throws {Error} If an error occurs while creating the room document.
+ */
+const createRoom = async (req, res) => {
+  const { name, description, region } = req.body;
+  const payload = { name, description, region };
+  try {
+    const roomData = await apiService.post("/rooms", payload);
+    const savedRoomData = await eventQuery.createRoom(roomData);
+    return res.status(200).json(savedRoomData);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't create room. Please try again later",
+    });
+  }
+};
+
+/**
+ * Retrieves all rooms that match the query parameters provided in the HTTP request query string.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The rooms data in JSON format.
+ * @throws {Error} If an error occurs while retrieving the rooms data.
+ */
+const getAllRooms = async (req, res) => {
+  const { enabled, hits, offset } = req.query;
```

Comment:
offset here is the id from which the rooms it will return.

May be we should try adding comments for body/params/query so that it's clear and helpful for others also, who will work on the project in the near future.